pkg/operator/operator/validations.go

```diff
@@ -744,3 +786,53 @@ func getValidationK8sResources() ([]kunstructured.Unstructured, *kresource.Quant
 
 	return virtualServices, maxMem, err
 }
+
+func validateOverridenImages(images, keys []string) error {
+	images = slices.RemoveEmpties(images)
+	keys = slices.RemoveEmpties(keys)
+
+	ECRImages := make([]bool, len(images))
+	for i, image := range images {
+		if regex.IsValidECR(image) {
+			ECRImages[i] = true
+		}
+	}
+
+	ECRAuthConfig := aws.ECRAuthConfig{}
+	dockerECRAuthConfig := dockertypes.AuthConfig{}
+	if slices.HasTrue(ECRImages) {
+		tokenOutput, err := aws.GetECRAuthToken()
```

Comment:
This runs in the operator, and currently we do not include ECR permissions on the list of required permissions [here](https://www.cortex.dev/cluster-management/security#operator). Do you know if we need any permissions other than `ecr:GetAuthorizationToken` to run the accessibility check? If not, here's what I'm thinking we could do:

* Add `ecr:GetAuthorizationToken` to the list of operator credentials we ask for
* **If the image is on ECR:** is it possible to distinguish between the repo not existing vs it existing and the IAM user not having access to it? If so, the former should be an error, whereas the latter I think we can ignore silently, since the nodes themselves have an IAM policy (that eksctl creates) which is used to pull the ECR images. We may want to check that the account ID for the ECR path matches that of the IAM user, since that will avoid a weird case where the IAM user has access to a different account's repo (the role the eksctl created will not have access to repos in other accounts). See `GetCachedAccountID()` in `lib/aws/sts.go` for getting the account ID.
  * an alternative to this is to use the AWS go client to check if the ECR image exists, rather than the docker client. The may give us more control over the error handling
* **If the image is not on ECR:** any error is an error that we should propagate

Reply:
This may not be necessary information for this code, but just FYI: the role that eksctl creates for pulling the images has the `AmazonEC2ContainerRegistryReadOnly` policy, which has these permissions:

```text
"Effect": "Allow",
"Action": [
    "ecr:GetAuthorizationToken",
    "ecr:BatchCheckLayerAvailability",
    "ecr:GetDownloadUrlForLayer",
    "ecr:GetRepositoryPolicy",
    "ecr:DescribeRepositories",
    "ecr:ListImages",
    "ecr:DescribeImages",
    "ecr:BatchGetImage",
    "ecr:GetLifecyclePolicy",
    "ecr:GetLifecyclePolicyPreview",
    "ecr:ListTagsForResource",
    "ecr:DescribeImageScanFindings"
],
"Resource": "*"
```

Reply:
Done!